docs/Polly-V8.md

```diff
@@ -0,0 +1,234 @@
+
+This API exposes the following classes and interfaces that can be used to create the resilience strategy:
+
+- `IResilienceStrategyBuilder`
+- `ResilienceStrategyBuilder`: concrete implementation of `IResilienceStrategyBuilder`.
+
+``` csharp
+public interface IResilienceStrategyBuilder
+{
+    ResilienceStrategyBuilderOptions Options { get; set; }
+
+    IResilienceStrategyBuilder AddStrategy(IResilienceStrategy strategy, ResilienceStrategyOptions? options = null);
+
+    IResilienceStrategyBuilder AddStrategy(Func<ResilienceStrategyBuilderContext, IResilienceStrategy> factory, ResilienceStrategyOptions? options = null);
+
+    IResilienceStrategy Create();
+}
+```
+
+To create a strategy you chain various extensions for `IResilienceStrategyBuilder` followed by the `Create` call:
+
+Single strategy:
+
+``` csharp
+var resilienceStrategy = new ResilienceStrategyBuilder().AddRetry().Create();
+```
+
+Pipeline of strategies:
+
+``` csharp
+var resilienceStrategy = new ResilienceStrategyBuilder()
+    .AddRetry()
+    .AddCircuitBreaker()
+    .AddTimeout(new TimeoutStrategyOptions() { ... })
+    .Create();
+```
+
+### Extensibility
+
+The resilience extensibility is simple. You just expose new extensions for `IResilienceStrategyBuilder` that use the `IResilienceStrategyBuilder.AddStrategy` methods.
+
+### Handling of different result types
+
+The resilience strategy can handle many different result types and exceptions as retry strategy sample demonstrates:
+
+``` csharp
+var options = new RetryStrategyOptions();
+options
+    .ShouldRetry
+    .Add<HttpResponseMessage>(m => m.StatusCode == HttpStatusCode.InternalServerError) // inspecting the result
+    .Add(HttpStatusCode.InternalServerError) // particular value for other type
+    .Add<MyResult>(v => v.IsError)
+    .Add<MyResult>((v, context) => IsError(context)) // retrieve data from context for evaluation
+    .AddException<InvalidOperationException>() // exceptions
+    .AddException<HttpRequestMessageException>() // more exceptions
+    .Add<MyResult>((v, context) => await IsErrorAsync(v, context)); // async predicates
+```
+
+In the preceding sample retry strategy handles 3 different types of results. This allows sharing of the retry strategy across the different result types. It is also possible to retrieve additional details from the `ResilienceContext` when handling the result.
+
+### Packages
+
+All packages have the same `Polly` root namespace.
+
+#### Polly.Abstractions
+
+Contains core primitives for consumers that just want to accept the core `IResilienceStrategy` in their APIs.
+
+**Dependencies:** None
+
+**Types**:
+
+- `IResilienceStrategy`
+- `ResilienceStrategyExtensions`
+- `Context`
+- `NullResilienceStrategy`
+- `DelegatingResilienceStrategy`
+
+#### Polly.Core
+
+Contains the builder for resilience strategies and core primitives that can be reused across various strategies.
+
+**Dependencies:**
+
+- `Poly.Abstractions`
+- `Microsoft.Extensions.Logging.Abstractions`
```

Comment:
We have added a lot of boilerplate code on top of Polly just to add telemetry. For us telemetry is a must as we need to track any failures and react on that. It would be great if all strategies have telemetry by default. 

If not enabled, it adds basically zero overheat. For this reason, I have included the `Microsoft.Extensions.Logging.Abstractions` in here. It's a core package that's very stable, because a LOT of modern libraries depend on it. We can also make our logger just to keep this dependency out, but I would discourage that.

Reply:
Yeah I can get the reasoning - I just know that the wider community can get a bit _"flaming pitchfork"_ over additional dependencies and which logging framework to use etc.

In a vacuum, I wouldn't mind adding it, just raising the point for discussion.